sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -862,6 +862,15 @@ expression
     : booleanExpression
     ;
 
+namedArgumentExpression
+    : key=identifier FAT_ARROW value=expression
```

Comment:
identifier can be quoted and unquoted
Shall we make it `IDENTIFIER`(unquoted) only?

Reply:
@gengliangwang After looking at the design doc, would it be fine to keep it like this?